### DISCUSSION TITLE
Add setWithParams and deleteWithParams to CacheAtom

### DISCUSSION THREAD

**Discussion by artalar** [reatom/reatom#796] [Ideas]
We need a way to set the cache manually. For example, we have an API which makes a `PUT`, `POST`, or _mutation_ call that returns new data which we can use without performing extra invalidation. The problem is that `cacheAtom` stores a complex data with some meta (`CacheRecord`) by keys with a specific semantics. We need a user-friendly way to update the cache by passing just a key and a new data.

```ts
export interface CacheAtom<T = any, Params extends any[] = unknown[]>
  extends MapAtom<unknown, CacheRecord<T, Params>> {
  /* NEW */setWithParams: Action<[params: Params, value: T]>
  /* NEW */deleteWithParams: Action<[params: Params]>
}
```

`WithParams` needed as `MapAtom` already have `set` and `delete` methods.

**Comment by BANOnotIT** [comment]
I don't really get what they should do and why they are needed?

**Comment by artalar** [comment]
@BANOnotIT description addded

**Comment by BANOnotIT** [comment]
Wouldn't it be better to have CacheAtom#invalidateParams instead of delete?

**Comment by artalar** [comment]
It should be additional feature, yes.

**Comment by artalar** [comment]
done https://github.com/artalar/reatom/commit/b53fe98b37e178735240b274336c29de6e3be1c5
